doc/how-to-run-in-windows.rst

```diff
@@ -0,0 +1,143 @@
+How to use Cabal in Windows
+===========================
+
+This document describes how to use Cabal in a Windows system. See the
+:ref:`Further reading` section for some other references that might provide some
+more explanations. For a TL;DR, jump to the :ref:`Complete configuration`.
+
+Install the Haskell environment
+-------------------------------
+
+The recommended way of setting up a Haskell environment in Windows is by using
+`GHCup <https://www.haskell.org/ghcup/>`_. Follow the steps outlined in its
+webpage to install at least GHC and Cabal. GHCup will install its own MSYS2
+system in your computer unless told not to do so, refer to `its documentation
+<https://www.haskell.org/ghcup/install/#windows_1>`_ for more information.

```

Comment:
As mentioned, I feel there should be some mention of Stack here, since it also comes with its own MSYS2 environment.

Reply:
Should this note be enough?

```rst
.. NOTE::
   Stack is another tool you can use to set up a Haskell environment. Stack
   can be installed on its own or via GHCup. See 
   `stack's webpage <https://docs.haskellstack.org/en/stable/>`_ and/or 
   `GHCup's section on stack integration <https://www.haskell.org/ghcup/guide/#stack-integration>`_,
   in particular the `Windows related subsection <https://www.haskell.org/ghcup/guide/#windows>`_.
```

Reply:
I’d be happy with this. We could put GHCup in a note, too. (Maybe even a single note for both of them?)

Reply:
Perhaps I should write this as normal text instead of a note. I feel like the section `Install the Haskell environment` should cover these directly, not as side-notes.

For now it says:
- Haskell uses MSYS2
- GHCup is recommended. Install GHC and Cabal. It might bootstrap MSYS2.
- Stack also exists. It can be installed via GHCup. It might bootstrap MSYS2.

Let me rephrase the paragraph a bit and push a commit (that we can squash later on)

Reply:
No, I think a note is correct, because this isn’t actually related to Cabal itself.

Reply:
I mean, how to get cabal (the executable) on Windows should be part of the Cabal docs no? 

Or do you prefer that this whole section is a note? I do think that mentioning MSYS2 and GHCup is relevant for cabal.

Reply:
> I mean, how to get cabal (the executable) on Windows should be part of the Cabal docs no?

Hmm… yeah, I guess this makes sense. Whether it’s in a note or not isn’t a hugely critical point.

Also, now that I think of it, Stack is a full replacement for Cabal. If you’re using Stack, you’re probably not going to be looking at Cabal docs! So there’s probably no need to mention it at all.

Reply:
I think it depends exactly on what @mpilgrem said:

> If its primary purpose is the use of Cabal (the tool), provided by the cabal-install package, on Windows, I would not expect Stack to feature. If its primary purpose is the use of Cabal (the library) on Windows, then Stack is one way to use that library:

I personally thought this as docs for cabal-the-tool, not cabal-the-library, but as you wanted Stack to appear I am ok with mentioning it. I'm fine either way.

Reply:
> I personally thought this as docs for cabal-the-tool, not cabal-the-library

Yeah, you’re right about this. I wasn’t thinking clearly when I made the suggestion. So the current text of the note is fine as-is.

Reply:
Let me push a commit right now, also including your suggestions below, and we can continue the discussion after it.

Reply:
Done, close this conversation if you are satisfied with the current contents.

Thanks!

Reply:
Like I said, I’m now convinced there’s no need to mention Stack. But now that it is mentioned, I don’t strongly feel that situation needs to be changed. either. I’ve been quibbling over this documentation for long enough!

(There appears to be no close buttons for this comment…)